lib/liquid/block_body.rb

```diff
@@ -114,7 +114,7 @@ def self.rescue_render_node(context, output, line_number, exc, blank_tag)
         when token.start_with?(TAGSTART)
           whitespace_handler(token, parse_context)
           unless token =~ FullToken
-            BlockBody.raise_missing_tag_terminator(token, parse_context)
```

Comment:
liquid-c still calls this if the invalid tag token isn't terminated with `%}`.

Reply:
That is true. Should we just change liquid-c to be consistent with liquid (and call `BlockBody.raise_missing_tag_terminator` instead of `yield`)? Should we also change how comments are handled because I think everything inside a comment block should be escaped?

Reply:
This doesn't have to be one or the other.  We can just check for `if token.end_with?('%}')` to only call `BlockBody.raise_missing_tag_terminator` only if the tag terminator is actually missing, which avoids breaking backwards compatibility with liquid templates and provides the more appropriate error message.

> Should we also change how comments are handled because I think everything inside a comment block should be escaped?

We have explicitly allowed tags with an invalid tag name (https://github.com/shopify/liquid/pull/256) so it seems very likely that this is relied upon.

Reply:
But the problem is that for a tag that looks like `{% comment %}{% " %}{% endcomment %}` is not matched on `FullToken` (and `token.end_with?('%}')` is true for `{% " %}`) in liquid so `BlockBody.raise_missing_tag_terminator` is called. But in liquid-c, it's yielded back to the caller (from the part you linked in the comment above).

Reply:
Exactly, so you want to fix that problem, but not introduce another inconsistency when the tag terminator is actually missing.  That's why I am recommending adding an `if token.end_with?('%}')` conditional here.

For example, `Liquid::Template.parse('{% foo')` should result in the syntax error `Liquid syntax error: Tag '{%' was not properly terminated with regexp: /\%\}/` and not `Liquid syntax error: Unknown tag '{%'` as the current change would change it to without liquid-c.  We could also use a test for the error messages for in this case, rather than just asserting that it results in a syntax error.

Reply:
An I see, that makes sense. Thanks for clarifying!